src/Common/src/System/Security/Cryptography/EccSecurityTransforms.cs

```diff
@@ -0,0 +1,409 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Security.Cryptography.Apple;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    internal sealed class EccSecurityTransforms : IDisposable
+    {
+        private SecKeyPair _keys;
+
+        public void Dispose()
+        {
+            _keys?.Dispose();
+            _keys = null;
+        }
+
+        internal int GenerateKey(ECCurve curve)
+        {
+            curve.Validate();
+
+            if (!curve.IsNamed)
+            {
+                throw new PlatformNotSupportedException(SR.Cryptography_ECC_NamedCurvesOnly);
+            }
+
+            int keySize;
+
+            switch (curve.Oid.Value)
+            {
+                // secp256r1 / nistp256
+                case "1.2.840.10045.3.1.7":
```

Comment:
It looks like S.S.C.Algorithms has only ever needed these three OIDs, and never got around to making an Oids.cs to drop them into.  Since there's source shared between Algorithms and the public implementation types (CNG, OpenSsl) I'll kick this can one more time and leave this as-is.